Userland/Libraries/LibJS/Runtime/AbstractOperations.cpp

```diff
@@ -176,4 +179,29 @@ ObjectEnvironmentRecord* new_object_environment(Object& object, bool is_with_env
     return global_object.heap().allocate<ObjectEnvironmentRecord>(global_object, object, environment_record);
 }
 
+// 19.2.1.1 PerformEval ( x, callerRealm, strictCaller, direct ), https://tc39.es/ecma262/#sec-performeval
+Value perform_eval(VM& vm, Value x, GlobalObject& caller_realm, CallerMode strict_caller, EvalMode direct)
```

Comment:
no need for the `VM` parameter here, we can easily get that via `auto& vm = global_object.vm();`